### DISCUSSION TITLE
How to match a dynamic part of a template string?

### DISCUSSION THREAD

**Discussion by HcySunYang** [ast-grep/ast-grep#1193] [Q&A]
Hi team,

I wanted to match the dynamic part of a template string, for example, given the following string:

```ts
const styles = css`
  color: ${token};
`
```

How to match the `${token}` part? Since it contains the `$` sign which is used as part of the Meta Variable's syntax, how do I escape it? BTW, I am using the JS API

**Comment by HerringtonDarkholme** [comment]
You can use the `template_substitution` kind. https://ast-grep.github.io/playground.html is pretty helpful

<img width="634" alt="image" src="https://github.com/ast-grep/ast-grep/assets/2883231/c904ac02-6890-42e4-a074-7d8976ba28f5">

**Reply by HerringtonDarkholme** [reply]
I guess you can use this rule

https://ast-grep.github.io/playground.html#eyJtb2RlIjoiQ29uZmlnIiwibGFuZyI6ImphdmFzY3JpcHQiLCJxdWVyeSI6IiRNRVRIT0QoXG4gICAgJEEsIC8vIENvbW1lbnRcbiAgICAkRlxuKSIsInJld3JpdGUiOiJtX21lbWJlck5ld05hbWUiLCJjb25maWciOiJydWxlOlxuICBraW5kOiB0ZW1wbGF0ZV9zdWJzdGl0dXRpb25cbiAgaW5zaWRlOlxuICAgIGtpbmQ6IHRlbXBsYXRlX3N0cmluZ1xuICAgIGluc2lkZTpcbiAgICAgIGtpbmQ6IGNhbGxfZXhwcmVzc2lvblxuICAgICAgaGFzOlxuICAgICAgICBmaWVsZDogZnVuY3Rpb25cbiAgICAgICAgcmVnZXg6IGNzc1xuICAgICIsInNvdXJjZSI6ImNvbnN0IHN0eWxlcyA9IGNzc2BcbiAgY29sb3I6ICR7dG9rZW59O1xuYCJ9

**Reply by HerringtonDarkholme** [reply]
Relevant doc for rules https://ast-grep.github.io/guide/rule-config.html

**Comment by HcySunYang** [comment]
@HerringtonDarkholme Thank you, very helpful
